test/zone_test.rb

```diff
@@ -48,6 +48,9 @@ def test_zone_has_config_with_provider
   end
 
   def test_zone_is_invalid_if_provider_does_not_support_record_type
+    supported_record_types = Provider::DynECT.record_types - ['ALIAS']
+    Provider::DynECT.stubs(:record_types).returns(supported_record_types)
```

Comment:
We need to think through how we want to model this since some zones can't support `ALIAS` records (💩  DynECT 💩 ). 

I'm currently leaning to making it an explicit config option for zones that belong to DynECT and adding a validation onto the zone. The alternative is we need to make sure there's a test that can run during CI to catch `ALIAS` records being added to unsupported zones.